src/server/lib/LiquidAutoCompleteRule.ts

```diff
@@ -0,0 +1,319 @@
+                return;
+            }
+            const hashValue = hash.value?.getText()
+            if (hashValue?.includes(AUTO_COMPLETE_PLACEHOLDER) && ['id', 'name', 'key'].includes(hashName)) {
+                suggestions.push(...getSuggestionsForEntity(PortalEntityNames.ENTITY_LIST, ctx, hashValue, hashName === 'id'))
+                return;
+            }
+        })
+        return suggestions
+    }
+}
+
+const webFormTagRule: ILiquidAutoCompleteRule = {
+    name:'webFormTag',
+    isValid: (liquidToken) => liquidToken.kind === TokenKind.Tag && (liquidToken as TagToken).name.toLowerCase() === PortalTags.WEBFORM,
+    priority: DEFAULT_TAG_PRIORITY,
+    apply: (liquidToken, ctx) => {
+        const suggestions: CompletionItem[] = []
+        const tokenizer = new Tokenizer((liquidToken as TagToken).args)
+        const hashes = tokenizer.readHashes();
+        hashes.forEach(hash => {
+            const hashName = hash.name.getText()
+            if (hashName?.includes(AUTO_COMPLETE_PLACEHOLDER)) {
+                suggestions.push(...WEB_FORM_ATTRIBUTES.map(key => {
+                    return {
+                        label: key,
+                        insertText: `${key}:`,
+                        kind: CompletionItemKind.Value
+                    } as CompletionItem
+                }))
+                return;
+            }
+            const hashValue = hash.value?.getText()
+            if (hashValue?.includes(AUTO_COMPLETE_PLACEHOLDER) && ['id', 'name', 'key'].includes(hashName)) {
+                suggestions.push(...getSuggestionsForEntity(PortalEntityNames.WEBFORM, ctx, hashValue, hashName === 'id'))
+                return;
+            }
+        })
+        return suggestions
+    }
+}
+
+const includeTagRule: ILiquidAutoCompleteRule = {
+    name: 'includeTag',
+    isValid: (liquidToken) => liquidToken.kind === TokenKind.Tag && (liquidToken as TagToken).name.toLowerCase() === PortalTags.INCLUDE,
+    priority: DEFAULT_TAG_PRIORITY,
+    apply: (tagToken, ctx) => {
+        const suggestions: CompletionItem[] = []
+        const tokenizer = new Tokenizer((tagToken as TagToken).args)
+        const value = tokenizer.readValue()
+        const templateName = value?.getText() || ''
+        if (templateName.includes(AUTO_COMPLETE_PLACEHOLDER)) {
+            suggestions.push(...getSuggestionsForEntity(PortalEntityNames.WEB_TEMPLATE, ctx, templateName))
+            return suggestions
+        }
+        const hashes = tokenizer.readHashes();
+        hashes.forEach(hash => {
+            const hashName = hash.name.getText().toLowerCase()
+            const hashValue = hash.value?.getText()
+            if (hashName === PortalAttributeNames.SNIPPET_NAME && hashValue?.includes(AUTO_COMPLETE_PLACEHOLDER) && templateName.replace(/['"]/g, '').toLowerCase() === PortalAttributeNames.SNIPPET) {
+                suggestions.push(...getSuggestionsForEntity(PortalEntityNames.CONTENT_SNIPPET, ctx, hashValue))
```

Comment:
Would be better to use a constant to maintain regex - /['"]/g

Reply:
Done